0000-template.md

```diff
@@ -48,6 +49,7 @@ Why should we *not* do this?
 - Why is this design the best in the space of possible designs?
 - What other designs have been considered and what is the rationale for not choosing them?
 - What is the impact of not doing this?
+- If this is a language proposal, could this be done in a library or macro instead? Does the proposed change make Rust code easier or harder to read, understand, and maintain?
```

Comment:
We may also consider IDE assistence (in RLS, e.g.) in improving ergonomics if the change in language could not improve lanugage consistency, or reducing language complexity. We do own RLS, so IDEs should also constitute as a component when we view "Rust" as a product.

Reply:
IDE assistance plays a part for sure, particularly to improve writing flow, but remember that not everyone uses an IDE. In my view it should be possible to log into a headless server with only `nano`, or a bare bones `vim` installation, and modify some small piece of Rust code and recompile for example.

Therefore, I think that readability and maintenance of code should not be much based on IDEs. For example, in my view, duplicating code with an IDE as a substitute for parametric polymorphism and traits in the past would not have been a good trade-off, and going forward, similar expressiveness-boosts in the type system should not be substituted with IDEs.

Reply:
I think you misunderstood my point. The readability and maintenance of the code relies on stability of the language. If something that can be done outside of the language, we probably should consider do it outside of the language because it reduces the learning burden of the user, and alleviate the gradual language bloat we see in also most long-lived major languages.

Reply:
No, I understood that point. Auto-completion can play a part, but if auto-completion adds large amounts of boilerplate from repetitive tasks, then it harms readability (for the person who is looking at a git PR for example...), maintenance and suggests that the language is not up to the task, or that the language hasn't given library authors the ability to automate.

Using Vim as your every day editor is very different from logging into a headless server and using a `nano` which has zero syntax highlighting and no plugins for Rust. It is just a text editor, it has no IDE facilities.

I'm going to avoid the discussion on Ok-wrapping here.